### ISSUE TITLE
T150 USB product string missing

### ISSUE THREAD

**Issue by thomasa88** [scarburato/t150_driver#23]
I'm trying to get my T150 to work with Project Cars 2 and Automobilista, using Steam Proton. ( https://github.com/ValveSoftware/Proton/issues/4819 )

I don't think this is related to my problem, but when I checked lsusb, the product name is picked up from my Prusa printer.

    lsusb -d 044f:b677
    
    Bus 001 Device 098: ID 044f:b677 ThrustMaster, Inc. Original Prusa i3 MK3

Looking closer in, I can see that idProduct shows no text.

```
lsusb -v -d 044f:b677

  idVendor           0x044f ThrustMaster, Inc.
  idProduct          0xb677 
  bcdDevice            1.00
```

Xbox Controller for comparison

```
lsusb -d 045e:02d1
Bus 001 Device 099: ID 045e:02d1 Microsoft Corp. Xbox One Controller

lsusb -d 045e:02d1 -v
  idVendor           0x045e Microsoft Corp.
  idProduct          0x02d1 Xbox One Controller
  bcdDevice            1.01
```

First I though that the firmware might be behaving badly, but I think the strings from the firmware appear in the following values (the actual strings are only fetched when running with "sudo")
```
  iManufacturer           1 Thrustmaster
  iProduct                2 Thrustmaster T150RS
  iSerial                 0 
```

[lsusb-xbox-sudo.txt](https://github.com/scarburato/t150_driver/files/6470961/lsusb-xbox-sudo.txt)
[lsusb-xbox.txt](https://github.com/scarburato/t150_driver/files/6470962/lsusb-xbox.txt)
[lsusb-prusa-sudo.txt](https://github.com/scarburato/t150_driver/files/6470963/lsusb-prusa-sudo.txt)
[lsusb-prusa.txt](https://github.com/scarburato/t150_driver/files/6470964/lsusb-prusa.txt)
[lsusb-v-t150.txt](https://github.com/scarburato/t150_driver/files/6470965/lsusb-v-t150.txt)
[lsusb-v-t150-sudo.txt](https://github.com/scarburato/t150_driver/files/6470966/lsusb-v-t150-sudo.txt)

**Comment by thomasa88** [comment]
It turns out that the value is missing from the Linux USB ID database.

Website:
http://www.linux-usb.org/usb-ids.html

File on Ubuntu:
/var/lib/usbutils/usb.ids

I will try editing usb.ids and, if it works, I will submit the ID for review to linux-usb.org

https://gist.github.com/m4ver1Kc/90462139e5ab0ab5e11e42d6779b5289 indicates that the product should be  "Thrustmaster T150RS"

I will see if I can read a value from a Windows computer as well, just to be sure.

**Comment by thomasa88** [comment]
Turns out that the linux-usb.org webpage is mostly broken, but it seems that the usb.ids list still gets updated. I will try to send a patch to the robot, but I'm not sure if one must be registered first, somehow.

USB ids are nowadays read from "hwdb" in systemd, but they seem to fetch the USB IDs from the usb.ids file.


/etc/udev/hwdb.d/90-thrustmaster-t150.hwdb
```
usb:v044FpB677*
 ID_MODEL_FROM_DATABASE=T150 Racing Wheel
```

I added a local hwdb entry in this way (Ubuntu 20.04):
```
sudo systemd-hwdb update

lsusb -d 044f:b677
Bus 001 Device 101: ID 044f:b677 ThrustMaster, Inc. T150 Racing Wheel

lsusb -d 044f:b677 -v
  idVendor           0x044f ThrustMaster, Inc.
  idProduct          0xb677 T150 Racing Wheel

```
Based on this guide:
https://wiki.archlinux.org/title/Map_scancodes_to_keycodes

Upstream systemd hwdb: https://github.com/thomasa88/systemd/tree/main/hwdb.d
